build/patches/Always-ask-for-permissions.patch

```diff
@@ -0,0 +1,291 @@
+---
+ .../java/res/xml/privacy_preferences.xml      |  5 ++
+ .../download/FileAccessPermissionHelper.java  |  2 +-
+ .../PermissionUpdateInfoBarDelegate.java      | 10 ++-
+ ...mission_update_infobar_delegate_android.cc | 11 ++-
+ .../strings/android_chrome_strings.grd        |  8 ++
+ .../strings/android/browser_ui_strings.grd    |  4 +
+ .../PermissionParamsListBuilder.java          |  4 +
+ .../AndroidPermissionRequester.java           | 81 ++++++++++++++++++-
+ .../PermissionDialogController.java           |  3 +
+ 9 files changed, 121 insertions(+), 7 deletions(-)
+
+diff --git a/chrome/android/java/res/xml/privacy_preferences.xml b/chrome/android/java/res/xml/privacy_preferences.xml
+--- a/chrome/android/java/res/xml/privacy_preferences.xml
++++ b/chrome/android/java/res/xml/privacy_preferences.xml
+@@ -70,6 +70,11 @@
+         android:title="@string/open_external_links_incognito_title"
+         android:summary="@string/open_external_links_incognito_summary"
+         android:defaultValue="false" />
++    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
++        android:key="ask_permissions_to_user"
++        android:title="@string/ask_permissions_to_user_title"
++        android:summary="@string/ask_permissions_to_user_summary"
++        android:defaultValue="true" />
+ 
+     <PreferenceCategory
+         android:key="services_category"
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/download/FileAccessPermissionHelper.java b/chrome/android/java/src/org/chromium/chrome/browser/download/FileAccessPermissionHelper.java
+--- a/chrome/android/java/src/org/chromium/chrome/browser/download/FileAccessPermissionHelper.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/download/FileAccessPermissionHelper.java
+@@ -65,7 +65,7 @@ public class FileAccessPermissionHelper {
+                                 && grantResults[0] == PackageManager.PERMISSION_GRANTED,
+                         null));
+ 
+-        AndroidPermissionRequester.showMissingPermissionDialog(
++        AndroidPermissionRequester.showMissingPermissionDialogUnchecked(
+                 ApplicationStatus.getLastTrackedFocusedActivity(),
+                 org.chromium.chrome.R.string.missing_storage_permission_download_education_text,
+                 ()
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/infobar/PermissionUpdateInfoBarDelegate.java b/chrome/android/java/src/org/chromium/chrome/browser/infobar/PermissionUpdateInfoBarDelegate.java
+--- a/chrome/android/java/src/org/chromium/chrome/browser/infobar/PermissionUpdateInfoBarDelegate.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/infobar/PermissionUpdateInfoBarDelegate.java
+@@ -23,6 +23,11 @@ import java.util.Collections;
+ import java.util.HashSet;
+ import java.util.Set;
+ 
++import org.chromium.components.permissions.AndroidPermissionRequester;
++
++// remove this comment when moved to
```

Comment:
This discussion should be in the PR description not in the comments.

When would the move happen? v96?

Reply:
> This discussion should be in the PR description not in the comments.

ok

> When would the move happen? v96?

I don't know, I still haven't figured out (well) how to see it (I know, you already explained it to me :)

Reply:
I do not understand what has to be moved and who decides when

Reply:
fixed in https://github.com/bromite/bromite/pull/1606/commits/e5a236efbc5c2fa74667089fa3d4c508563fc4aa

`PermissionUpdateInfoBarDelegate.java` was removed in v96